### PR TITLE
Update secrets-backend.md

### PR DESCRIPTION
**Pull request by fritz-astronomer**
Customer reports that the import path changed and the instructions were confusing

### CODE REVIEW

File: astro/secrets-backend.md [hunk 1]
```diff
@@ -465,21 +465,24 @@ By default, this setup requires that you prefix any secret names in Key Vault wi
```
Comment (jwitz) [inline astro/secrets-backend.md:465]: I think the note about Git is more appropriate here.

**Review by jwitz — CHANGES_REQUESTED** [review]
@fritz-astronomer Is there a reason we're not telling users to set the environment variables as secret in Astro any more?

**Comment by fritz-astronomer** [comment]
> @fritz-astronomer Is there a reason we're not telling users to set the environment variables as secret in Astro any more?

History, no good reason. Let's pick something and standardize 🤷

File: astro/secrets-backend.md [hunk 2]
```diff
@@ -455,7 +455,7 @@ Add the following environment variables to your `.env` file:
 AZURE_CLIENT_ID="<your-client-id>" # Found on App Registration page > 'Application (Client) ID'
 AZURE_TENANT_ID="<your-tenant-id>" # Found on App Registration page > 'Directory (tenant) ID'
 AZURE_CLIENT_SECRET="<your-client-secret>" # Found on App Registration Page > Certificates and Secrets > Client Secrets > 'Value'
-AIRFLOW__SECRETS__BACKEND=airflow.providers.microsoft.azure.secrets.azure_key_vault.AzureKeyVaultBackend
+AIRFLOW__SECRETS__BACKEND=airflow.providers.microsoft.azure.secrets.key_vault.AzureKeyVaultBackend
```
Comment (fritz-astronomer) [inline astro/secrets-backend.md:458]: https://github.com/apache/airflow/blob/main/airflow/providers/microsoft/azure/secrets/key_vault.py#L37
Just proving to myself that this is the correct path, cuz I had originally just guessed 😅

**Review by jwitz — APPROVED** [review]

**Pull request merged**